tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,278 @@
+import math
+import pytest
+from decimal import Decimal
+import datetime
+from datetime import timezone, timedelta
+import trino
+
+
+@pytest.fixture
+def trino_connection(run_trino):
+    _, host, port = run_trino
+
+    yield trino.dbapi.Connection(
+        host=host, port=port, user="test", source="test", max_attempts=1
+    )
+
+
+def test_boolean_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="false", python=False) \
+        .add_field(sql="true", python=True) \
+        .add_field(sql="cast(null AS BOOLEAN)", python=None) \
+        .execute()
+
+
+def test_tinyint_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(127 AS TINYINT)", python=127) \
+        .add_field(sql="cast(-128 AS TINYINT)", python=-128) \
+        .add_field(sql="cast(42 AS TINYINT)", python=42) \
+        .add_field(sql="cast(-42 AS TINYINT)", python=-42) \
+        .add_field(sql="cast(null AS TINYINT)", python=None) \
+        .execute()
+
+
+def test_smallint_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(32767 AS SMALLINT)", python=32767) \
+        .add_field(sql="cast(-32768 AS SMALLINT)", python=-32768) \
+        .add_field(sql="cast(null AS SMALLINT)", python=None) \
+        .execute()
+
+
+def test_int_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(2147483647 AS INTEGER)", python=2147483647) \
```

Comment:
all the `add_field` calls which test min/max values could use a comment like `# min value in Trino`.

Also are there data types where the range supported by Trino is outside the range supported by the Python type? If so we should test them explicitly (also with comments like `# min value in Trino, not representable in Python` or something like that).

Reply:
The comments actually don't mix well with the `\`

Reply:
Very unfortunate. 😢 Let's live with it.

Reply:
"Unresolved" for the second question. I think an example is `DATE` where Trino has wider range than Python?

Reply:
Yes, `DATE` indeed. But didn't you say to drop the temporal test types for now?

Reply:
Yes. Let's drop dates. But are there other types with such a case? DATE was the only example I know of.

Reply:
It's my impression that Python barely has any limit when it comes to any type of number.

Reply:
I was thinking of the other direction. i.e. types in Trino which have wider range that Python. Seems there are none. We can revisit when we find any. Resolved.